app/Http/Requests/Album/SetAlbumCopyrightRequest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Http\Requests\Album;
+
+use App\Contracts\Http\Requests\HasBaseAlbum;
+use App\Contracts\Http\Requests\HasCopyright;
+use App\Contracts\Http\Requests\RequestAttribute;
+use App\Http\Requests\BaseApiRequest;
+use App\Http\Requests\Traits\Authorize\AuthorizeCanEditAlbumTrait;
+use App\Http\Requests\Traits\HasBaseAlbumTrait;
+use App\Http\Requests\Traits\HasCopyrightTrait;
+use App\Http\RuleSets\Album\SetAlbumCopyrightRuleSet;
+
+class SetAlbumCopyrightRequest extends BaseApiRequest implements HasBaseAlbum, HasCopyright
+{
+	use HasBaseAlbumTrait;
+	use HasCopyrightTrait;
+	use AuthorizeCanEditAlbumTrait;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function rules(): array
+	{
+		return SetAlbumCopyrightRuleSet::rules();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function processValidatedValues(array $values, array $files): void
+	{
+		$this->album = $this->albumFactory->findBaseAlbumsOrFail(
```

Comment:
you may need to do this in two steps. 
get the value in a temporary variable to type it with `/** @var Album ... */` and then assign to `$this->album`

Not the most convenient, I know.